pages/error.js

```diff
@@ -40,6 +41,7 @@ function Error() {
       setValues(values);
     }
     fetchData();
+    setError(JSON.parse(router.query.error));
```

Comment:
the error comes URL encoded, which is then parsed into an object here, to later be stringified with proper formatting, see L64 below